manager/src/main/java/org/openremote/manager/energy/ForecastSolarService.java

```diff
@@ -162,144 +177,246 @@ protected static void initClient() {
+                            assetPredictedDatapointService.updateValue(electricityProducerSolarAsset.getId(), ElectricityProducerSolarAsset.POWER.getName(), powerKiloWatt, dateTime);
+                        }
+                    });
                 }
-            } catch (Throwable e) {
-                if (e.getCause() != null && e.getCause() instanceof IOException) {
-                    LOG.log(Level.SEVERE, "Exception when requesting forecast solar data", e.getCause());
-                } else {
-                    LOG.log(Level.SEVERE, "Exception when requesting forecast solar data", e);
+                rulesService.fireDeploymentsWithPredictedDataForAsset(electricityProducerSolarAsset.getId());
+            } else {
+                StringBuilder message = new StringBuilder("Unknown");
+                if (response != null) {
+                    message.setLength(0);
+                    message.append("Status ");
+                    message.append(response.getStatus());
+                    message.append(" - ");
+                    message.append(response.readEntity(String.class));
                 }
+                LOG.warning("Request failed: " + message);
             }
+        } catch (Throwable e) {
+            if (e.getCause() != null && e.getCause() instanceof IOException) {
+                LOG.log(Level.SEVERE, "Exception when requesting forecast solar data", e.getCause());
+            } else {
+                LOG.log(Level.SEVERE, "Exception when requesting forecast solar data", e);
+            }
+        }
+    }
+
+    protected void updateSolarForecastAttribute(ElectricityProducerSolarAsset electricityProducerSolarAsset) {
+        Optional<Double> lat = electricityProducerSolarAsset.getAttribute(Asset.LOCATION).flatMap(attr -> attr.getValue().map(GeoJSONPoint::getY));
+        Optional<Double> lon = electricityProducerSolarAsset.getAttribute(Asset.LOCATION).flatMap(attr -> attr.getValue().map(GeoJSONPoint::getX));
+        Optional<Integer> pitch = electricityProducerSolarAsset.getPanelPitch();
+        Optional<Integer> azimuth = electricityProducerSolarAsset.getPanelAzimuth();
+        Optional<Double> kwp = electricityProducerSolarAsset.getPowerExportMax();
+
+        if (lat.isEmpty() || lon.isEmpty() || pitch.isEmpty() || azimuth.isEmpty() || kwp.isEmpty()) {
+            return;
+        }
+
+        // Get solar forecast data-points for current 15 minute interval
+        long currentTimeMillis = timerService.getCurrentTimeMillis();
+        long startTimeMillis = currentTimeMillis - currentTimeMillis % (15 * 60000);
+        long endTimeMillis = currentTimeMillis - currentTimeMillis % (15 * 60000) + 15 * 60000;
+        AssetDatapointAllQuery assetDatapointQuery = new AssetDatapointAllQuery(startTimeMillis, endTimeMillis);
```

Comment:
Would it be possible to simplify these data/time calculations by using a LocalDateTime object instead? 

I see that `LocalDateTime` can also be used as `AssetDatapointAllQuery` constructor argument:

```java
public AssetDatapointAllQuery(LocalDateTime fromTime, LocalDateTime toTime) {
```